kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -495,6 +499,56 @@ public static Optional<Tuple2<Protocol, Set<TableFeature>>> autoUpgradeProtocolB
     }
   }
 
+  /**
+   * Adds features from currentProtocol based on overrides in properties.
+   *
+   * <p>Overrides are specified using a key in th form {@linkplain
+   * #PROPERTIES_FEATURE_OVERRIDE_PREFIX} + {featureName}. (e.g. {@code
+   * delta.feature.icebergWriterCompatV1}). The value should be "supported" to add the feature.
+   * Currently, removing values is not handled.
+   *
+   * @return An updated properties map and protocol if there are overrides present and the features
+   *     are not in {@code currentProtocol} and the current protocol can be upgraded to support the
+   *     new features.
+   * @throws KernelException if the feature name for the override is invalid or the value is not
+   *     equal to "supported".
+   */
+  public static Optional<Tuple2<Protocol, Metadata>> updateProtocolWithFeaturePropertyOverrides(
```

Comment:
Don't we want to return something regardless (except in the error case). By that I mean -- don't return an optional?

case 1) we have updated the protocol. return the updated protocol and the filtered table properties
case 2) we have NOT updated the protocol. return the original protocol and ALSO RETURN the FILTERED table properties